samcli/local/docker/container.py

```diff
@@ -150,11 +155,14 @@ def create(self):
         if self._env_vars:
             kwargs["environment"] = self._env_vars
 
-        kwargs["ports"] = {self.RAPID_PORT_CONTAINER: ("127.0.0.1", self.rapid_port_host)}
+        kwargs["ports"] = {self.RAPID_PORT_CONTAINER: (self._container_host, self.rapid_port_host)}
 
         if self._exposed_ports:
             kwargs["ports"].update(
-                {container_port: ("127.0.0.1", host_port) for container_port, host_port in self._exposed_ports.items()}
+                {
+                    container_port: (self._container_host, host_port)
```

Comment:
Should we resolve the (potential) domain name before? I can't see domain names being mentioned in the [docs](https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.run).

Reply:
Yes we should resolve the domain name before passing into ports binding. Trying to figure out what is the right ip it should bind to when running in a container. Resolved `host.docker.internal` doesn't work in this case.

Reply:
Haven't figured out why resolved `host.docker.internal` doesn't work here. An idea just came to my mind. Whenever we use local Docker host, it should bind to `127.0.0.1` while the host name could be `localhost`, `host.docker.internal` etc. When using a remote Docker host, we just use the ip of Docker host. So maybe we should use 2 options?

Reply:
Overall, this looks good, what are the scenarios we have tested?

Reply:
Tested 3 scenarios and all work well:
1. Run `sam local` in local
2. Run `sam local` in a container with Docker socket binding.
3. Run `sam local` in a container with Docker running in the other container.

However I had a talk with Chris and we are thinking to add another option `--container-host-interface`. Because docker port binding can only use ip address. And the interface and container host don't match sometimes.(For example, when we use `host.docker.internal` as  container host, we would like bind ports to `127.0.0.1`) So ideally we need 2 options to cover all use case.